Docs GridLayout (#365)
telerik/blazor-docs: components/gridlayout/overview.md

    </GridLayoutItems>
</TelerikGridLayout>
````

>caption The result from the code snippet above

![overview of the GridLayout functionality](images/gridlayout-overview-basic-example.png121)

>caption Component namespace and reference

````CSHTML
<TelerikGridLayout @ref="@GridLayoutRef">
    <GridLayoutColumns>
        <GridLayoutColumn Width="200px"></GridLayoutColumn>
    </GridLayoutColumns>
    <GridLayoutRows>
    </GridLayoutRows>
    <GridLayoutItems>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 1
            </div>
        </GridLayoutItem>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 2
            </div>
        </GridLayoutItem>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 3
            </div>
        </GridLayoutItem>
    </GridLayoutItems>
</TelerikGridLayout>

@code {
    Telerik.Blazor.Components.TelerikGridLayout GridLayoutRef { get; set; }
}
````

## Features

The GridLayout offers the following features:

* `Class` - the CSS class that renders on the main wrapping element of the component.

> * `Width`- takes a CSS lenght unit that determines how wide the GridLayout is. See the [Dimensions]({%slug common-features/dimensions%}) article for more details on what units you can use and how dimensions in percent work.

Reviewer: ```suggestion
* `Width`- takes a CSS length unit that determines how wide the GridLayout is. See the [Dimensions]({%slug common-features/dimensions%}) article for more details on what units you can use and how dimensions in percent work.
```

---
title: Overview
page_title: GridLayout Overview

> description: Overview of the GridLayout for Blazor.

Reviewer: Use different keywords in the different meta elements.

```suggestion
description: Introduction to the GridLayout for Blazor.
```

slug: gridlayout-overview
tags: telerik,blazor,gridlayout,overview
published: True
position: 0
---

# GridLayout Overview

The <a href="https://www.telerik.com/blazor-ui/gridlayout" target="_blank">Blazor GridLayout component</a> allows you to arrange the contents of the component in rows and columns in a grid structure. 

#### In This Article:

* [Basics](#basics)

* [Features](#features)

> ## Basics

Reviewer: In this section we explain how rows, columns and items are defined. It becomes clear that the number of columns should match the number of items. However, it doesn't become clear if rows are required or not.

> #### To use a Telerik GridLayout for Blazor

Reviewer: We are skipping one heading level here (## -> ####). Is a heading needed at all?

```suggestion
To use a Telerik GridLayout for Blazor:
```

1. Declare the `<TelerikGridLayout>` tag 

> 1. To define columns in the GridLayout add one or more instances of the `<GridLayoutColumn>` to the `<GridLayoutColumns>` collection.

Reviewer: ```suggestion
1. To define columns in the GridLayout, add one or more `<GridLayoutColumn>` tags to the `<GridLayoutColumns>` collection.
```

>     * The number of `<GridLayoutColumn>` should be the same as the number of the `<GridLayoutItem>`.

Reviewer: ```suggestion
    * The number of `<GridLayoutColumn>`s should match the number of `<GridLayoutItem>`s.
```

> 1. To define rows in the GridLayout add one or more instances of the `<GridLayoutRow>` to the `<GridLayoutRows>` collection.

Reviewer: ```suggestion
1. To define rows in the GridLayout, add one or more `<GridLayoutRow>` tags to the `<GridLayoutRows>` collection.
```

> 1. Provide one or more instances of the `<GridLayoutItem>` to the `<GridLayoutItems>` collection.

Reviewer: ```suggestion
1. Define one or more `<GridLayoutItem>` tags in the `<GridLayoutItems>` collection.
```

        <GridLayoutColumn Width="200px"></GridLayoutColumn>
        <GridLayoutColumn Width="200px"></GridLayoutColumn>
    </GridLayoutColumns>
    <GridLayoutRows>
        <GridLayoutRow Height="50px"></GridLayoutRow>
    </GridLayoutRows>
    <GridLayoutItems>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 1
            </div>
        </GridLayoutItem>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 2
            </div>
        </GridLayoutItem>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 3
            </div>
        </GridLayoutItem>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 4
            </div>
        </GridLayoutItem>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 5
            </div>
        </GridLayoutItem>
    </GridLayoutItems>
</TelerikGridLayout>
````

>caption The result from the code snippet above

![overview of the GridLayout functionality](images/gridlayout-overview-basic-example.png121)

> >caption Component namespace and reference

Reviewer: This second snippet brings little value, but expands the article length. It can be merged with the first one.

Author: This is a common practice across our documentation, to add a snippet that showcases the namespace and reference. I would rather leave it there.

````CSHTML
<TelerikGridLayout @ref="@GridLayoutRef">
    <GridLayoutColumns>
        <GridLayoutColumn Width="200px"></GridLayoutColumn>
    </GridLayoutColumns>
    <GridLayoutRows>
    </GridLayoutRows>
    <GridLayoutItems>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 1
            </div>
        </GridLayoutItem>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 2
            </div>
        </GridLayoutItem>
        <GridLayoutItem>
            <div style="border: 1px gray solid;">
                item 3
            </div>
        </GridLayoutItem>
    </GridLayoutItems>
</TelerikGridLayout>

@code {
    Telerik.Blazor.Components.TelerikGridLayout GridLayoutRef { get; set; }
}
````

## Features

> The StackLayout offers the following features:

Reviewer: We are introducing a StackLayout term, which has not been defined. Is this a typo? Is this a feature of the GridLayout? Is this a separate component? How does it fit in the above information and snippets?

Author: Indeed, it is a typo. Will be fixed with the rest of the improvements based on your feedback

* `Class` - the CSS class that renders on the main wrapping element of the component.

> * `Width`- takes a CSS unit that determines how wide the GridLayout is. See the [Dimensions]({%slug common-features/dimensions%}) article for more details on what units you can use and how dimensions in percent work.

Reviewer: ```suggestion
* `Width`- takes a CSS length unit that determines how wide the GridLayout is. See the [Dimensions]({%slug common-features/dimensions%}) article for more details on what units you can use and how dimensions in percent work.
```

> * `ColumnSpacing` - controls the space between the columns in the GridLayout. See the [Dimensions]({%slug common-features/dimensions%}) article for more details on what units you can use and how dimensions in percent work.

Reviewer: We have already linked the Dimensions article.

```suggestion
* `ColumnSpacing` - controls the space between the columns in the GridLayout. Use a CSS length unit.
```

> * `RowSpacing` - controls the space between the rows in the GridLayout. See the [Dimensions]({%slug common-features/dimensions%}) article for more details on what units you can use and how dimensions in percent work.

Reviewer: ```suggestion
* `RowSpacing` - controls the space between the rows in the GridLayout. Use a CSS length unit.
```

> * `HorizontalAlign` - controls the alignment of the inner items in the GridLayout column based on the X axis. Takes a member of the `GridLayoutHorizontalAlign` enum:

Reviewer: Which are the inner items and which are the "outer" items?

Author: The inner items are all <GridLayoutItem>s defined. They are inner to the component
